### PR TITLE
feat: Add arma3-windows support.

### PR DESCRIPTION
**Pull request by Mattzi**
This should(tm) be working.
We need to test this in a dev environment @P4sca1.
Full feature support of the old image

### CODE REVIEW

File: arma3-windows/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM mcr.microsoft.com/windows/servercore:ltsc2022
+
+MAINTAINER mattis.kraemer (at) einfach-gaming.de
+
+RUN NET USER /add arma3
+RUN NET LOCALGROUP "Administrators" "arma3" /add
+
+user arma3
```
Comment (P4sca1) [inline arma3-windows/Dockerfile:8]: PleAse rename to USER

File: arma3-windows/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+FROM mcr.microsoft.com/windows/servercore:ltsc2022
+
+MAINTAINER mattis.kraemer (at) einfach-gaming.de
```
Comment (P4sca1) [inline arma3-windows/Dockerfile:3]: The MAINTAINER feature in Dockerfiles is deprecated.

File: arma3-windows/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+COPY "./entrypoint.ps1" "C:/entrypoint.ps1"
+ 
+COPY "./X3DAudio1_7.dll" "C:/temp/X3DAudio1_7.dll"
+COPY "./XAPOFX1_5.dll" "C:/temp/XAPOFX1_5.dll"
+
+COPY "./vcredist_x64.exe" "C:/temp/vcredist_x64.exe"
+
+WORKDIR "C:/arma3server"
+Volume "C:/arma3server"
```
Comment (P4sca1) [inline arma3-windows/Dockerfile:18]: Please rename to VOLUME

File: arma3-windows/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
+FROM mcr.microsoft.com/windows/servercore:ltsc2022
+
+MAINTAINER mattis.kraemer (at) einfach-gaming.de
+
+RUN NET USER /add arma3
+RUN NET LOCALGROUP "Administrators" "arma3" /add
```
Comment (P4sca1) [inline arma3-windows/Dockerfile:6]: Is it required that the arma3 user is an admin? I would prefer running the server with least privileges.

File: arma3-windows/Readme.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+* **update_validate** Same like update but will also validate the files. Recommended for the initial installation of the server.
+* **start** Only start the arma3 server without installing or updating.
+
+## Data persistence
+Game server data is kept in `C:/arma3server`.
+By default a volume will be auto-created which will persist the game server data across server restarts.
+When you re-create the container, a new volume is created and you can't access the old data unless you manually mount the old volume.
+See https://docs.docker.com/storage/volumes/ for more information.
+
+To persist the game server data on the host filesystem, use `-v /absolute-path/on/host:C:/arma3server` when creating the docker container.
```
Comment (P4sca1) [inline arma3-windows/Readme.md:18]: Does the mounting work that way because of the double colons?

File: arma3-windows/entrypoint.ps1 [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+        $latestRpt = Get-ChildItem "logfiles" -Filter "*.rpt" `
+                    | Sort-Object LastWriteTime -Descending `
+                    | Select-Object -First 1
+    }
+    catch { }
+
+    return $latestRpt
+}
+
+function Compare-SteamCredentials {
```
Comment (P4sca1) [inline arma3-windows/entrypoint.ps1:45]: Maybe Check-SteamCredentials is a more appropriate name?

File: arma3-windows/Dockerfile (hunk 4 above)
Comment (Mattzi) [reply arma3-windows/Dockerfile:6]: I need to test this, might be possible to remove admin permissions.
But I might be forced to change the volume location

**Comment by P4sca1** [comment]
Is there an official download link to the .exe and .dll files so that we do not have to store them in the repo but can copy them via http into the image during build?

Also, we need to add a GitHub workflow for the windows build.

File: arma3-windows/entrypoint.ps1 (hunk 6 above)
Comment (Mattzi) [reply arma3-windows/entrypoint.ps1:45]: Sure, In the beginning I tried to use the Accepted Verbs for Powershell Functions but those are quite limited
https://docs.microsoft.com/de-de/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.2

**Comment by Mattzi** [comment]
> Is there an official download link to the .exe and .dll files so that we do not have to store them in the repo but can copy them via http into the image during build?
> 
> Also, we need to add a GitHub workflow for the windows build.

May be possible if we can extract a rar in the build.
Vcredist could be problematic because the microsoft mirror is just valid for a limited amount of time after clicking download

File: arma3-windows/Dockerfile (hunk 4 above)
Comment (Mattzi) [reply arma3-windows/Dockerfile]: So after testing we have permission problems with every part of the entrypoint and dockerfile afterwards. We could probably do it anyways but as everybody knows, the windows permission system is horrible to work with

Comment (P4sca1) [reply arma3-windows/Dockerfile]: Do you run an ARMA server as Administrator normally on Windows? (e.g. using TADST).
If that's not the case, it should be possible to make it run without admin privileges. Maybe it's just about choosing the right folder where the data is placed.

**Pull request merged**